ArtOfIllusion/src/artofillusion/image/ImageMap.java

```diff
@@ -251,5 +253,5 @@ public void setName(String newName)
       
       which reconstructs an image from its serialized representation. */
 
-  public abstract void writeToStream(DataOutputStream out) throws IOException;
+  public abstract void writeToStream(DataOutputStream out, Scene scene) throws IOException;
```

Comment:
Changing the Public API of one of our core classes. I think I see why, but do we know if this would break anything?

@peastman

Reply:
As AOI has no Plugin API to connect other image formats I think this is no-breaking change

Reply:
It's impossible to guarantee it won't break something, because we don't know what other code might be calling this.  The question is how important we think that is.

Reply:
This is part of the serialization code, though. Not sure I've ever run across a Plugin or Script that wanted to serialize *part* of a scene, Not sure I even can think of a realistic use case for that... It's normally called when writing out an entire scene.

I was more concerned about Plugins adding new ImageMap types, but Makiam is right that this is not currently possible (At least without overriding parts of AOI core code.)

Your call.

Reply:
I agree it's pretty unlikely to break things.  Probably not worth worrying about.

Reply:
Sounds good to me.